applications/DEMApplication/python_scripts/DEM_properties_measure_utility.py

```diff
@@ -0,0 +1,215 @@
+        
+        if type == "fabric_tensor":
+            if self.DEM_parameters["ContactMeshOption"].GetBool():
+                measured_fabric_tensor = self.ContactElementGlobalPhysicsCalculator.CalculateFabricTensorWithinSphere(self.contact_model_part, radius, [center_x, center_y, center_z])
+                measured_fabric_tensor = np.array(measured_fabric_tensor)
+                deviatoric_tensor = 4 * (measured_fabric_tensor - 1/3 * np.eye(3)) 
+                second_invariant_of_deviatoric_tensor = (0.5 * np.sum(deviatoric_tensor * deviatoric_tensor))**0.5
+                eigenvalues, eigenvectors = np.linalg.eig(measured_fabric_tensor)
+                return eigenvalues, second_invariant_of_deviatoric_tensor, measured_fabric_tensor
+            else:
+                raise Exception('The \"ContactMeshOption\" in the [ProjectParametersDEM.json] should be [True].')
+            
+        if type == "stress_tensor":
+            if self.DEM_parameters["PostStressStrainOption"].GetBool() and self.DEM_parameters["ContactMeshOption"].GetBool():
+                measure_sphere_volume = 4.0 / 3.0 * math.pi * radius * radius * radius
+                Lx = domain_size[0]
+                Ly = domain_size[1]
+                Lz = domain_size[2]
+                total_stress_tensor = self.ContactElementGlobalPhysicsCalculator.CalculateTotalStressTensorWithinSphere(self.contact_model_part, radius, [center_x, center_y, center_z], Lx, Ly, Lz)
+                averaged_stress_tensor = np.array(total_stress_tensor) / measure_sphere_volume
+                return averaged_stress_tensor
+            else:
+                raise Exception('The \"PostStressStrainOption\" and \"ContactMeshOption\" in the [ProjectParametersDEM.json] should be [True].')
+            
+        if type == "unbalanced_force":
+            if self.DEM_parameters["ContactMeshOption"].GetBool():
+                measured_unbalanced_force = self.ContactElementGlobalPhysicsCalculator.CalculateUnbalancedForceWithinSphere(self.spheres_model_part, self.contact_model_part, radius, [center_x, center_y, center_z])
+                return measured_unbalanced_force
+            else:
+                raise Exception('The \"ContactMeshOption\" in the [ProjectParametersDEM.json] should be [True].')
+        
+        if type == "conductivity_tensor":
+            if self.DEM_parameters["ContactMeshOption"].GetBool():
+                measure_sphere_volume = 4.0 / 3.0 * math.pi * radius * radius * radius
+                total_tensor = np.empty((3, 3))
+                total_tensor[:] = 0.0
+                total_contact_number = 0
+                angles_xy = []
+                angles_xz = []
+                angles_yz = []
+
+                for element in self.contact_model_part.Elements:
+                    x_0 = element.GetNode(0).X
+                    x_1 = element.GetNode(1).X
+                    y_0 = element.GetNode(0).Y
+                    y_1 = element.GetNode(1).Y
```

Comment:
This can be optimized by using `element.GetNode(0).Coordinates()`

Reply:
besides, moving it in the future to c++ would be nice

Reply:
Maybe @jgonzalezusua could have a look at this part